apollo-livedata-support/src/main/java/com/apollographql/apollo/livedata/LiveDataResponse.kt

```diff
@@ -0,0 +1,74 @@
+package com.apollographql.apollo.livedata
+
+import com.apollographql.apollo.api.Error
+import com.apollographql.apollo.api.Response
+
+/**
+ * The LiveDataResponse class maps [Response] to [LiveDataResponse].
+ * There are Success, Failure, UnsuccessfulResponse, Complete sealed type classes.
+ */
+
+@Suppress("unused")
+sealed class LiveDataResponse<out T> {
+
+    /**
+     * API Success response class.
+     *
+     * [data] is an Optional. (There are responses without data)
+     */
+    class Success<out T>(response: Response<T>) : LiveDataResponse<T>() {
+        val operation = response.operation()
+        val data: T? = response.data()
+        val dependentKeys: Set<String> = response.dependentKeys()
+        val fromCache: Boolean = response.fromCache()
+
+        override fun toString() = "[ApiResponse.Success: $data]"
+    }
+
+    /**
+     * API Failure response class.
+     *
+     * ## Throw Exception case.
+     * Communication is not successful or an exception related to the model deserialization occurs.
+     *
+     * ## API Network format error case.
+     * API communication conventions do not match or applications need to handle exceptions or errors.
+     */
+    class Failure<out T>(private val error: Exception, response: Response<out T>? = null) : LiveDataResponse<T>() {
```

Comment:
How about to define 2 types of errors: one when smth related to request execution (basically in case of any ApolloException) and errors returned by GraphQL server (response has some errors)

Reply:
As you can see, the `companion object` creates two type of Failures about `ApolloException` and 
`response.errors`.  The `of` method will create Failure class If `response.hasErrors()` is true.
But the Failure class is handling `Exception` now. I think it is too ambiguous. So It will fixed to

Reply:
As you can see, the `companion object` creates two types of Failures about `ApolloException` and 
`response.errors`.  The `of` method will create Failure class If `response.hasErrors()` is true.
But the Failure class is handling just `Exception` now. I think it is too ambiguous. So It will be fixed to `ApolloException` and `toString` override.

Reply:
But there is an issue with that. We have 2 fields that indicates about the error. I believe that these 2 errors are mutual exclusive (I hope they are as in case of exception we can't really have response, on the other hand in case of GraphQL errors we can't have exception). 

That's the reason why we have in `com.apollographql.apollo.ApolloCall.Callback` 2 calls:
```
    /**
     * Gets called when GraphQL response is received and parsed successfully. Depending on the
     * {@link ResponseFetcher} used with the call, this may be called multiple times. {@link #onCompleted()}
     * will be called after the final call to onResponse.
     *
     * @param response the GraphQL response
     */
    public abstract void onResponse(@NotNull Response<T> response);

    /**
     * Gets called when an unexpected exception occurs while creating the request or processing the response.
     * Will be called at most one time. It is considered a terminal event. After called,
     * neither {@link #onResponse(Response)} or {@link #onCompleted()} will be called again.
     */
    public abstract void onFailure(@NotNull ApolloException e);
```
In case of GraphQL server error `onResponse` will be called.

But they are mutual exclusive. So my suggestion is to make `Failure` to be sealed class with 2 subclasses: `ApolloException` and `GraphQLError` (not sure that names are good enough, might be better).

Reply:
I agree to it.  It seems to need specifying the Failures(exception case, error case).
It will be fixed. Thank you so much!